lib/jekyll/url.rb

```diff
@@ -67,32 +67,32 @@ def generate_url(template)
 
     def generate_url_from_hash(template)
       @placeholders.inject(template) do |result, token|
-        break result if result.index(':').nil?
+        break result if result.index(":").nil?
         if token.last.nil?
-          # Remove leading '/' to avoid generating urls with `//`
-          result.gsub(/\/:#{token.first}/, '')
+          # Remove leading "/" to avoid generating urls with `//`
+          result.gsub(%r!/:#{token.first}!, "")
         else
           result.gsub(/:#{token.first}/, self.class.escape_path(token.last))
         end
       end
     end
 
     def generate_url_from_drop(template)
-      template.gsub(/:([a-z_]+)/.freeze) do |match|
-        replacement = @placeholders.public_send(match.sub(':'.freeze, ''.freeze))
+      template.gsub(/:([a-z_]+)/) do |match|
+        replacement = @placeholders.public_send(match.sub(":".freeze, "".freeze))
```

Comment:
If we are gonna do `"".freeze` everywhere, we need do like Liquid does and create EMPTY_STRING constant, because this is nasty and overwhelming. /cc @jekyll/core